src/commands/delete/delete-worker.ts

```diff
@@ -0,0 +1,68 @@
+import { parentPort, threadId } from 'node:worker_threads';
+
+import { fsa } from '@chunkd/fs';
+import { WorkerRpc } from '@wtrpc/core';
+
+import { logger } from '../../log.ts';
+import { ConcurrentQueue } from '../../utils/concurrent.queue.ts';
+import { registerCli } from '../common.ts';
+import type { DeleteContract, DeleteContractArgs, DeleteStats } from './delete-rpc.ts';
+
+const Q = new ConcurrentQueue(10);
+
+/** Current log id */
+let currentId: string | null = null;
+
+export const worker = new WorkerRpc<DeleteContract>({
```

Comment:
Worker threads are generally if your are hitting a single threaded CPU limit (which it does with hashing), 

deletes shouldn't need use all the CPU Cores on the machine, you could greatly simplify this by removing all the worker logic